### PR TITLE
implement the TLS Transport Parameters

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: internal/handshake/params_negotiator.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	var foundInitialMaxData bool
+	var foundInitialMaxStreamID bool
+	var foundIdleTimeout bool
+
+	for _, p := range params {
+		switch p.Parameter {
+		case initialMaxStreamDataParameterID:
+			foundInitialMaxStreamData = true
+			if len(p.Value) != 4 {
+				return fmt.Errorf("wrong length for initial_max_stream_data: %d", len(p.Value))
```
Comment (lucas-clemente) [inline internal/handshake/params_negotiator.go]: Should we include the expected len in the error message?

File: internal/handshake/params_negotiator.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+// newParamsNegotiator creates a new connection parameters manager
+func newParamsNegotiator(pers protocol.Perspective, v protocol.VersionNumber, params *TransportParameters) *paramsNegotiator {
+	h := &paramsNegotiator{}
+	h.perspective = pers
+	h.version = v
+	h.init(params)
+	return h
+}
+
+func (h *paramsNegotiator) SetFromTransportParameters(params []transportParameter) error {
```
Comment (lucas-clemente) [inline internal/handshake/params_negotiator.go]: Do we have to lock the mutex here?

Comment (marten-seemann) [reply internal/handshake/params_negotiator.go]: Yeah, I missed that. Same for `GetTransportParameters`.

File: internal/handshake/params_negotiator.go (hunk 1 above)
Comment (marten-seemann) [reply internal/handshake/params_negotiator.go]: We can do that.

**Pull request merged**